ragas-eval-flow/README.md

```diff
@@ -0,0 +1,92 @@
+# Ragas Evaluation Flow
+
+This evaluation flow uses the [ragas evaluation framework](https://github.com/explodinggradients/ragas) to calculate evaluation metrics.
+
+> *Ragas is a framework that helps you evaluate your Retrieval Augmented Generation (RAG) pipelines. Ragas provides you with the tools based on the latest research for evaluating LLM-generated text to give you insights about your RAG pipeline. Ragas can be integrated with your CI/CD to provide continuous checks to ensure performance.*
+
+See [Ragas documentation](https://docs.ragas.io/en/latest/) for more information.
+
+## Inputs
+
+- **question**: original user prompt
+- **answer**: answer generated by the model
+- **context**: context provided to the model in a prompt
+- **ground_truth**: expected answer
+
+## Output
+
+A batch run of the flow creates an output that contains a JSON object of the following shape **for each input in the
+data set**:
+
+```json
+{
+    "results": {
+        "faithfulness": 1.0,
+        "answer_relevancy": 0.9750335489,
+        "context_recall": 0.6666666667,
+        "context_precision": 0.9999999999,
+        "harmfulness": 0.0
+    },
+    "line_number": 0
+}
+```
+
+## Metrics
+
+A batch run of the flow also **aggregates metrics across the entire data set** and logs them using the Prompt Flow
+`log_metric()` function.
+
+The aggregated metrics of a certain flow run can be seen using the following command:
+
+```bash
+pf run show-metrics --name <ragas evaluation flow run name>
+```
+
+Sample output:
+
+```json
+{
+    "faithfulness": 1.0,
+    "answer_relevancy": 0.84,
+    "context_recall": 0.89,
+    "context_precision": 0.67,
+    "harmfulness": 0.0
+}
+```
+
+## How run the flow
+
+> **Note:** the flow needs *all* of the following environment variables to be provided in order to configure Ragas
+> library. The values below are just examples and need to be replaced with your own values.
+
+### Batch run against another flow run
+
+To run the `ragas-eval-flow` against a named run of the `chat-with-patents` flow, run the following command:
+
+```bash
+pf run create -f ragas_eval_run.yaml --run <chat with patents run name> \
```

Comment:
this gives me RateLimitError although the source run (on data.jsonl) only contains 3 lines. Any way to test?